reward/src/lib.rs

```diff
@@ -0,0 +1,546 @@
+        let (provider, parser) = ProviderOptions::from(parser)?;
+        let (signer, mut parser) = SignerOptions::from(parser)?;
+        let mut range = None;
+        let mut dry = false;
+        let mut amount = None;
+        let mut token = None;
+        let mut strategy = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("dry") => {
+                    dry = true;
+                }
+                Value(val) if range.is_none() => {
+                    range = val.into_string().ok();
+                }
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Long("token") => {
+                    token = parser.value()?.parse().ok();
+                }
+                Long("amount") => {
+                    let amt = parser
+                        .value()?
+                        .into_string()
+                        .map_err(|_| anyhow!("Can't parse --amount value"))?;
+                    amount = U256::from_dec_str(&amt).ok();
+                }
+                Long("strategy") => {
+                    strategy = parser.value()?.parse().ok();
+                }
+                _ => return Err(anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((
+            Options {
+                range,
+                dry,
+                amount,
+                token,
+                strategy,
+                provider,
+                signer,
+            },
+            vec![],
+        ))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let rt = tokio::runtime::Runtime::new()?;
+    let provider = ethereum::provider(options.provider)?;
+    let signer_opts = options.signer;
+    let (wallet, provider) = rt.block_on(term::ethereum::get_wallet(signer_opts, provider))?;
+    let signer: Arc<_> = SignerMiddleware::new(provider, wallet).into();
+    let profile = profile::default()?;
+    let storage = profile::read_only(&profile)?;
+    let repo = git::Repository::open(Path::new("."))?;
+
+    let mut revwalk = repo.revwalk()?;
+    let range = options
+        .range
+        .or_else(|| {
+            if !repo.tag_names(Some("rad/reward")).ok()?.is_empty() {
+                return Some("rad/reward..".to_string());
+            }
+            None
+        })
+        .ok_or_else(|| anyhow!("Can't determine the range of commits"))?;
+    revwalk.push_range(&range)?;
+
+    // from identities in local monorepo build a map of: fp -> ens
+    let mut ssh_to_ens: HashMap<String, _> = HashMap::new();
+    identities::any::list(&storage)?
+        .filter_map(|identity| {
+            if let Ok(Person(person)) = identity {
+                let peer_ids = person
+                    .delegations()
+                    .iter()
+                    .map(|key| {
+                        let peer_id: PeerId = key.to_owned().into();
+                        peer_id
```

Comment:
You can put the type in the closure and not need this assignment: `|key| -> PeerId { ... }`